frontend/server/src/main/java/org/pytorch/serve/wlm/BatchAggregator.java

```diff
@@ -103,20 +105,20 @@ public void sendError(BaseModelRequest message, String error, int status) {
                 String requestId = req.getRequestId();
                 Job job = jobs.remove(requestId);
                 if (job == null) {
-                    logger.error("Unexpected job: " + requestId);
+                    logger.error("Unexpected job in sendError(): " + requestId);
                 } else {
                     job.sendError(status, error);
                 }
             }
             if (!jobs.isEmpty()) {
                 jobs.clear();
-                logger.error("Not all jobs get response.");
+                logger.error("Not all jobs got an error response.");
             }
         } else {
             // Send the error message to all the jobs
             for (Map.Entry<String, Job> j : jobs.entrySet()) {
                 String jobsId = j.getValue().getJobId();
-                Job job = jobs.remove(jobsId);
+                Job job = jobs.get(jobsId);
```

Comment:
@msaroufim do we have any test to repro the batch aggregator issue?

Reply:
@msaroufim [ModelServerTest.java#L1071](https://github.com/pytorch/serve/blame/master/frontend/server/src/test/java/org/pytorch/serve/ModelServerTest.java#L1071) is a frontend integration test case which is called during frontend build.

Test test_huggingface_bert_batch_inference is an end 2 end integration which is called by regression test.

So they are different level test. Usually, a java test case should be added into dir frontend/server/src/test if there are frontend source code changes. Currently frontend misses a batch positive test case.

It is fine if you prefer adding batch end2end test cases in regression test. Then, I suggest you add both positive and negative test cases at here. Otherwise, the problem at regression for batch use case is as same as current Java frontend test (ie. miss a test case).